activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1002,13 +1002,19 @@ def invert_where! # :nodoc:
 
     # Checks whether the given relation is structurally compatible with this relation, to determine
     # if it's possible to use the #and and #or methods without raising an error. Structurally
-    # compatible is defined as: they must be scoping the same model, and they must differ only by
-    # #where (if no #group has been defined) or #having (if a #group is present).
+    # compatible is defined as: the given relation must be scoping the same model, and it must differ
+    # from the current relation only by #where (if no #group has been defined) or #having (if a #group is present).
```

Comment:
> it must differ from the current relation only by #where

Although this is really only true for multi value methods, I couldn't figure out a succint way of phrasing that...